packages/react-scripts/config/paths.js

```diff
@@ -117,3 +118,40 @@ if (
   };
 }
 // @remove-on-eject-end
+
+module.exports.srcPaths = [module.exports.appSrc];
+
+// recursively find package dirs
+// -- omit things in node_modules, nested packages, and current app package
+const findPkgs = root =>
```

Comment:
Why do we need this? I would assume that if we look for packages, we only respect the ones specified in `workspaces` field.

I don’t think it’s a good idea to recursively traverse directories on startup. This can get expensive on large source trees.

Reply:
> Why do we need this?

TLDR; Ejected polyfills.js.
At first, srcPaths only contained appSrc and monorepo root; monorepo root was a catch-all for everything, actually didn't even need to include appSrc.  That worked until eject ... at which point polyfills.js gets created in the app and we start transpiling it, and it fails.  I think the failure was due to 'use strict' and I thought Babel 7 was supposed to handle that, but I didn't really look into the failure, just realized polyfills.js wasn't intended to be transpiled.  At that point, I thought it was probably better to be more explicit as to which paths are treated as CRA src and changed srcPaths to include each package path individually instead of the monorepo root.  I'm happy to revisit that decision and try to figure out how to transpile polyfills successfully or some other solution if you prefer.

> I would assume that if we look for packages, we only respect the ones specified in workspaces field.
> I don’t think it’s a good idea to recursively traverse directories on startup. This can get expensive on large source trees.

Yes, it can be done more efficiently and accurately by honoring the workspaces field.  I'll make that change, at which point I think this should be pretty optimized as it already stops recursing further when it finds a package.

Reply:
Both YW and LP (Lerna Packages ...config) are just globs that point to package.json files. You shouldn't need to recurse. I hope to abstract the [LP resolution algorithm](https://github.com/lerna/lerna/blob/9c466231553045161e80607e16419aafc70a533a/src/PackageUtilities.js#L50-L88) into its own package soon.

Reply:
Nice, that would be really helpful to us!

Reply:
[Yarn's algorithm](https://github.com/yarnpkg/yarn/blob/00c4b0f3199c8eb2b6cc23e20d33faddb993f1a5/src/config.js#L647-L660) is similar.  I was going to try to simplify and optimize a bit here, mainly by preventing unnecessarily searching deeper into package dirs (e.g., for '**'), but I suppose that's a fool's errand.  I'll just copy the LP resolution algorithm above for now, I think that's what you're suggesting.  From this use case, a Lerna/Yarn CLI for finding packages might be better than using a separate finder package, in order to keep the finder consistent with Lerna/Yarn in use.  Thanks!